ui-framework-documentation.md

```diff
@@ -76,34 +89,85 @@ sub provide {
 }
 ```
 
-main.pm then calls a scheduled [Test Module](#test-module) that is using
-the Distribution to access its components through Controller layer.
+main.pm calls a scheduled [Test Module](#test-module) using [declarative scheduling](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/declarative-schedule-doc.md)
+(please don't schedule modules in main*.pm directly !) and uses the [Distribution](https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/lib/Distribution)
+to access its components through the [Controller](#controller) layer.
 
 >  Important: Test Module must be inherited from opensusebasetest or one
 >  of its children to have an access to the Distribution.
-
 ```perl
 use parent 'opensusebasetest';
 
 sub run {
     my $partitioner = $testapi::distri->get_partitioner_controller();
+    $partitioner->some_method_defined_in_controller();
+}
+```
+The method get_partitioner_controller calls the right controller according to the product version,
+eg for Tumbleweed:
+
+
+```perl
+package Distribution::Opensuse::Tumbleweed;
+use Installation::Partitioner::LibstorageNG::v4_3::ExpertPartitionerController;
+
+sub get_expert_partitioner {
+    return Installation::Partitioner::LibstorageNG::v4_3::ExpertPartitionerController->new();
 }
 ```
+In this example, the proper controller is v4.3. The class re-uses common parts from older versions of the product.
```

Comment:
the controller is not just 'v4.3', it needs the full name, because we can have many controllers with the same name in very different folders.

Reply:
changed